File: javascript/DataPacket.js
const QuantumObject = require('./QuantumObject');
const QuantumCollapseException = require('./QuantumCollapseException');

class DataPacket extends QuantumObject {
    analyze() {
        let newValue = this.Stability - 5;
        if (newValue <= 0) throw new QuantumCollapseException(this.ObjectID);
        this.Stability = newValue;
        console.log("Data decryption successful. Content read.");
    }
}
module.exports = DataPacket;

File: csharp/DataPacket.cs
using System;

namespace NDP_Odev4
{
    internal class DataPacket : QuantumObject
    {
        public DataPacket(string id, int stability, int dangerLevel) 
            : base(id, stability, dangerLevel) { }

        public override void Analyze()
        {
            int newValue = Stability - 5;
            if (newValue <= 0) throw new QuantumCollapseException(ObjectID);

            Stability = newValue;
            Console.WriteLine("Data decryption successful. Content read.");
        }
    }
}